File: 7_kyu/DigitDigit.py
def square_digits(num):
    num = int(num)
    accu = 0
    shift = 1
    while num > 0:
        d = num % 10
        num //= 10
        d2 = d*d
        accu += d2 * shift
        shift *= 10 if d2 < 10 else 100
    return accu

File: 7_kyu/DigitDigit.r
suppressMessages(library(purrr))

# Returns a number pasted together from the decimal digits of num squared.
square_digits <- function(num)
  reduce( .x = digits(num), .f = function(accu,d) d*d + accu*ifelse(d>3,100,10), .init=0 )

# Converts number n into the value of its decimal digits.         
digits <- function(n) as.integer(strsplit(as.character(n),"")[[1]])
